gateway-util-common/src/main/java/org/apache/knox/gateway/util/ConcurrentSessionVerifier.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.knox.gateway.util;
+
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class ConcurrentSessionVerifier {
+    private static Set<String> privilegedUsers = new HashSet<>();
+    private static Set<String> nonPrivilegedUsers = new HashSet<>();;
+    private static int privilegedUserConcurrentSessionLimit = 3;
+    private static int nonPrivilegedUserConcurrentSessionLimit = 2;
+    private static Map<String, Integer> concurrentSessionCounter = new HashMap<>();
```

Comment:
The name suggests this is not thread safe, should this be `ConcurrentHashMap` ?

Reply:
Yes changed it to ConcurrentHashMap.